browser/permissions/permission_manager_browsertest.cc

```diff
@@ -0,0 +1,242 @@
+    EXPECT_EQ(sub_request_origins[i],
+              permission_request_manager->Requests()[i]->GetOrigin());
+  }
+
+  // Test ignoring request.
+  permission_request_manager->IgnoreEthereumPermissionRequests();
+  EXPECT_TRUE(observer->IsRequestsFinalized());
+  EXPECT_TRUE(!observer->IsShowingBubble());
+  EXPECT_TRUE(IsEthereumPendingRequestsEmpty());
+
+  for (size_t i = 0; i < addresses.size(); i++) {
+    EXPECT_EQ(host_content_settings_map()->GetContentSetting(
+                  sub_request_origins[i], GetLastCommitedOrigin(),
+                  ContentSettingsType::BRAVE_ETHEREUM),
+              ContentSetting::CONTENT_SETTING_ASK);
+  }
+
+  observer->Reset();
+  permission_manager()->RequestPermissions(
+      types, web_contents()->GetMainFrame(), origin, true, base::DoNothing());
+
+  content::RunAllTasksUntilIdle();
+  EXPECT_TRUE(permission_request_manager->IsRequestInProgress());
+  EXPECT_TRUE(observer->IsShowingBubble());
+  EXPECT_FALSE(IsEthereumPendingRequestsEmpty());
+
+  // Check sub-requests are created as expected.
+  EXPECT_EQ(permission_request_manager->Requests().size(), addresses.size());
+  for (size_t i = 0; i < permission_request_manager->Requests().size(); i++) {
+    EXPECT_EQ(permission_request_manager->Requests()[i]->GetRequestType(),
+              RequestType::kBraveEthereum);
+    EXPECT_EQ(sub_request_origins[i],
+              permission_request_manager->Requests()[i]->GetOrigin());
+  }
+
+  // Test accepting request with one of the address.
+  permission_request_manager->AcceptEthereumPermissionRequests(
+      std::vector<std::string>{addresses[1]});
+  std::vector<ContentSetting> expected_settings(
+      {ContentSetting::CONTENT_SETTING_ASK,
+       ContentSetting::CONTENT_SETTING_ALLOW});
+  EXPECT_TRUE(observer->IsRequestsFinalized());
+  EXPECT_TRUE(!observer->IsShowingBubble());
+  EXPECT_TRUE(IsEthereumPendingRequestsEmpty());
+
+  for (size_t i = 0; i < addresses.size(); i++) {
+    EXPECT_EQ(host_content_settings_map()->GetContentSetting(
+                  sub_request_origins[i], GetLastCommitedOrigin(),
+                  ContentSettingsType::BRAVE_ETHEREUM),
+              expected_settings[i]);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(PermissionManagerBrowserTest, GetCanonicalOrigin) {
+  const GURL& url = https_server()->GetURL("/empty.html");
+  ui_test_utils::NavigateToURL(browser(), url);
+
+  std::vector<std::string> addresses = {
+      "0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8A",
+      "0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8B"};
+  GURL origin;
+  ASSERT_TRUE(brave_wallet::GetConcatOriginFromWalletAddresses(
+      GetLastCommitedOrigin(), addresses, &origin));
+
+  EXPECT_EQ(origin, permission_manager()->GetCanonicalOrigin(
+                        ContentSettingsType::BRAVE_ETHEREUM, origin,
+                        GetLastCommitedOrigin()))
+      << "GetCanonicalOrigin should return requesting_origin for Ethereum "
+         "permission.";
+}
+
```

Comment:
can we add a test which ensures nothing explode when a tab with an active request is closed?

Reply:
Added in https://github.com/brave/brave-core/pull/9443/commits/094d14e7430abfeeb54d636bb7eba7bf05a2842e